Remove code notifications

KarinaMendez2 opened the pull request.

Commit:
modified selector to return only one notification

Commit:
moved switch into a function
to render based on type

Commit:
Merge remote-tracking branch 'origin/master' into remove_code_notifications

Commit:
Merge remote-tracking branch 'origin/master' into remove_code_notifications

Commit:
initial solutions for notifications display, created a component
for all notifications, re organized elements to be inside new wrapper component,
modified selector to only return
one notification at a time

Commit:
Merge remote-tracking branch 'origin/master' into remove_code_notifications

TomWoodward commented on src/app/content/components/Content.tsx:
```diff
@@ -47,18 +47,6 @@ const Background = styled.div`
 `;
 
-// tslint:disable-next-line:variable-name
-const ContentNotifications = styled(Notifications)`
```
we need this here because the top offset is different in the `content` module than outside of it (because of the book banner and toolbar)

TomWoodward commented on src/app/notifications/components/Notifications.tsx:
```diff
@@ -1,70 +1,48 @@
-interface Props extends React.HTMLProps<HTMLDivElement> {
-  notifications: AnyNotification[];
+interface Props {
+  notifications: AnyNotification;
```
```suggestion
  notification: AnyNotification;
```

TomWoodward commented on src/app/notifications/components/Notifications.tsx:
```diff
@@ -1,70 +1,48 @@
-        default:
-          return null;
+const renderNotification = (notification: AnyNotification) => {
+  if (notification) {
```
notification isn't nullable in the signature, do you need this condition?

TomWoodward commented on src/app/notifications/components/NotificationsWrapper.tsx:
```diff
@@ -0,0 +1,80 @@
+import { inlineDisplayBreak } from '../theme';
+
+interface Props {
+  mobileExpanded?: boolean;
```
this is a prop that is only relevant in the content module

TomWoodward commented on src/app/notifications/components/NotificationsWrapper.tsx:
```diff
@@ -0,0 +1,80 @@
+
+export class NotificationsWrapper extends Component<Props> {
+    public render() {
+        return <Wrapper>
```
the idea is that we're trying to remove this wrapper and move all the css into the notifications themselves

Commit:
redistributed styled and elements to get rid of wrapper
snaps updated, z-index added

Commit:
Merge remote-tracking branch 'origin/master' into remove_code_notifications

Commit:
renamed variables, removed
console logs, snaps updated,
removed duplicated css

Commit:
indent with 2 spaces

Commit:
removed unused webkit sticky

Commit:
moved css from wrapper to
card Body for position sticky
to work, added className to
the other notification components, snaps updated

Commit:
merge conflicts

TomWoodward commented on src/app/content/components/Content.tsx:
```diff
@@ -42,7 +42,7 @@ const Background = styled.div`
 
 // tslint:disable-next-line:variable-name
-const ContentNotifications = styled(Notifications)`
+export const ContentNotifications = styled(Notifications)`
```
why do you need to export this?

TomWoodward commented on src/app/developer/components/Home.tsx:
```diff
@@ -4,7 +4,7 @@ import Footer from '../../components/Footer';
 import { bodyCopyRegularStyle, H1 } from '../../components/Typography';
 import { contentWrapperMaxWidth } from '../../content/components/constants';
-import DisplayNotifications from '../../notifications/components/Notifications';
+import NotificationCard from '../../notifications/components/Notifications';
```
coulda left this name as `DisplayNotification`

TomWoodward commented on src/app/notifications/components/AcceptCookies.tsx:
```diff
@@ -9,24 +9,31 @@ import { doAcceptCookies } from '../acceptCookies';
-    <ButtonGroup>
-      <FormattedMessage id='i18n:notification:cookies:ok'>
-        {(txt) => <Button variant='primary' onClick={dismiss}>{txt}</Button>}
+class AcceptCookies extends Component<Props> {
```
why change this to a class component?

TomWoodward commented on src/app/notifications/selectors.ts:
```diff
@@ -22,7 +22,7 @@ const messagePriority = [
 export const notificationsForDisplay = createSelector(
```
please make sure the type for this includes `| undefined`

TomWoodward commented on src/app/developer/components/Notifications.tsx:
```diff
@@ -24,6 +25,9 @@ const Notifications = ({updateAvailable, error, sendMessages}: Props) => {
   return <Panel title='Notifications'>
     <ButtonGroup expand={false}>
+
+      <Button onClick={acceptCookies} >accept cookies</Button>
```
👍

Commit:
removed unused export,
renamed variable, changed
class back into constant, removed spaces for lint

Commit:
renaming selector in singular,
added undefined on return type for notification display
selector

Commit:
fixed import name for notification
for display spec

Commit:
removed first attempt of returning undefined and
added undefined as type

TomWoodward commented on src/app/notifications/components/UpdatesAvailable.tsx:
```diff
@@ -1,29 +1,29 @@
-    window.location.reload(true);
-  }
-};
+export default class UpdatesAvailable extends Component<{className?: string}> {
```
why did you change this to a class component?

Commit:
removing class

Commit:
space
